bundles/org.openhab.binding.vigicrues/src/main/java/org/openhab/binding/vigicrues/internal/handler/VigiCruesHandler.java

```diff
@@ -0,0 +1,141 @@
+import java.time.ZonedDateTime;
+import java.util.Arrays;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.measure.Unit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.i18n.TimeZoneProvider;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.vigicrues.internal.VigiCruesConfiguration;
+import org.openhab.binding.vigicrues.internal.json.OpenDatasoftResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+import tec.uom.se.unit.Units;
+
+/**
+ * The {@link VigiCruesHandler} is responsible for querying the API and
+ * updating channels
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public class VigiCruesHandler extends BaseThingHandler {
+    private static final String URL = OPENDATASOFT_URL + "?dataset=vigicrues&sort=timestamp&q=";
+    private static final int TIMEOUT_MS = 30000;
+    private final Logger logger = LoggerFactory.getLogger(VigiCruesHandler.class);
+
+    // Time zone provider representing time zone configured in openHAB configuration
+    private final TimeZoneProvider timeZoneProvider;
+    private final Gson gson;
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private String queryUrl = "";
+
+    public VigiCruesHandler(Thing thing, TimeZoneProvider timeZoneProvider, Gson gson) {
+        super(thing);
+        this.timeZoneProvider = timeZoneProvider;
+        this.gson = gson;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing VigiCrues handler.");
+
+        VigiCruesConfiguration config = getConfigAs(VigiCruesConfiguration.class);
+        logger.debug("config station = {}", config.id);
+        logger.debug("config refresh = {} min", config.refresh);
+
+        updateStatus(ThingStatus.UNKNOWN);
+        queryUrl = URL + config.id;
+        refreshJob = scheduler.scheduleWithFixedDelay(this::updateAndPublish, 0, config.refresh, TimeUnit.MINUTES);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing the VigiCrues handler.");
+
+        ScheduledFuture<?> refreshJob = this.refreshJob;
+        if (refreshJob != null) {
+            refreshJob.cancel(true);
+        }
+        this.refreshJob = null;
```

Comment:
The only way it would be a different job is if `initialize()` was called concurrently with `dispose()`. And if that was the case then you would also need to take precautions to not overwrite an existing job without cancelling the old one.

Now if the framework never calls `initialize` or `dispose` concurrently then I think nulling the fields in `dispose` would be needless but harmless to do.

Reply:
So as I understand it is a common practice all over bindings, I leave it as it is